### ISSUE TITLE
SpanKnown vouches rows whose ParentIndex/Depth differ between builds, and the fuzzer cannot see it

### ISSUE THREAD

**Issue by richlander** [richlander/dotnet-inspect#3725]
## Summary

`DeclarationSpan.SpanKnown` is a claim about a row's **lines**. `Depth` and
`ParentIndex` are explicitly *not* covered by it, and the contract says so.
But the same paragraph also asserts a stronger property:

> Where the branches of a group disagree about which declaration encloses the
> text after the `#endif`, the group is refused outright rather than vouched for
> with one branch's nesting — that is what the opening-depth floor is for

**That sentence is false.** A vouched row can report a parent that is wrong in
half the builds, with every line identical in both.

## Reproduction

```csharp
#if A
class X
#else
class Y
#endif
{
    int dummy;
    class Nested { }
}
```

`Nested` is reported `SpanKnown=true`, `parent=Class:X`, `depth=1`, lines 8..8.
Without `A` the enclosing type is `Y`. Both configurations are valid C# and
`Nested`'s lines are identical in both, so no line-based gate can see it.

Remove `int dummy;` and the row is correctly refused. A bare `;` in its place
also reproduces it, so it is not specific to fields.

**Mechanism** (per round-11 review): a member inside the block calls
`ResetHeader`, and because its terminator sits outside the group
`crossesABranch` is false, so `ResetHeader` **assigns** `headerKnown = true`
rather than intersecting it. That discharges the conditional-header poison for
every later child. This is the same "knownness assigned rather than intersected"
defect as way 7, recurring at a different site.

## Why the fuzzer cannot see it

`Differ` and `DifferFromProduct` in `ConditionalRecoveryFuzzTests` compare
trivia, sig, sigEnd, body, end and attributes. **Neither compares `ParentIndex`
or `Depth` at all**, so a group that moves a vouched row's parent while leaving
every line identical is invisible by construction.

I prototyped closing this (add `ParentId`/`NestingDepth` to the oracle `Decl`,
compare both in each routine, spelling parent identity as `(Kind, Name)` because
the product's row list also carries rows written inside groups and the two lists
do not share an indexing). At seed 424242 x 20,000 it produced **23 flags**
immediately.

**16 of those are a harness artifact, not product defects.** The generator emits
a stray `public` line and attributes that can land immediately before a
namespace, and Roslyn's *parser* accepts `public namespace N;` and
`[Obsolete] namespace N;` while the language rejects both (CS1671). No such file
can compile, so "what encloses this row" has no answer to be right about.
Gating the nesting comparison on the absence of a modifier- or
attribute-bearing namespace drops it to **7 flags in compilable sources**.

Note this is *only* a problem for the nesting comparison — lines remain
well defined in those files, which is why the existing gates were unaffected.

## Scope note

Deliberately split out of #3680, which is scoped to the line claim and is being
kept landable. Round 11's line defect (the fourteenth way) is fixed there.

## Suggested work

1. Fix `ResetHeader` to intersect rather than assign, matching the way-7 fix.
2. Land the `ParentIndex`/`Depth` comparison in the fuzzer as a permanent gate,
   with the CS1671 scoping above.
3. Re-examine the remaining flags, which are all file-scoped namespaces reached
   through nested `#else`/`#elif` chains.
4. Replace the false sentence in the `SpanKnown` contract remarks with whatever
   survives, or mark the nesting property unverified per `AGENTS.md`.

## Evidence that valid re-parenting is otherwise refused

Probes A–M in round 11 confirm a conditional file-scoped namespace in the `#if`,
`#else`, or `#elif` branch is correctly refused, as is close-and-reopen at net
zero depth (the depth floor catches it because the close reaches below `d0`).
The known escape is a brace-less scope opener, which is way 3.

Found by adversarial review round 11 (Gemini 3.1 Pro).
